src/ConnectionAutomaton.cpp

```diff
@@ -286,7 +286,7 @@ void ConnectionAutomaton::reconnect(
 
 void ConnectionAutomaton::addStream(
     StreamId streamId,
-    std::shared_ptr<StreamAutomatonBase> automaton) {
+    yarpl::Reference<StreamAutomatonBase> automaton) {
```

Comment:
This is because the StreamAutomatonBase derived classes also derive from the yarpl::flowable::Subscriber/Subscription which inherit from yarpl::Refcounted. We need to use only one memory management mechanism. The common one is yarpl::Reference.